runtime/sema/type.go

```diff
@@ -1375,6 +1391,21 @@ func (*NumericType) CheckInstantiated(_ ast.HasPosition, _ common.MemoryGauge, _
 	// NO-OP
 }
 
+func (t *NumericType) EffectiveInterfaceConformanceSet() *InterfaceSet {
+	t.initializeEffectiveInterfaceConformanceSet()
+	return t.effectiveInterfaceConformanceSet
+}
+
+func (t *NumericType) initializeEffectiveInterfaceConformanceSet() {
+	t.effectiveInterfaceConformanceSetOnce.Do(func() {
+		t.effectiveInterfaceConformanceSet = NewInterfaceSet()
+
+		for _, conformance := range t.conformances {
+			t.effectiveInterfaceConformanceSet.Add(conformance)
+		}
+	})
+}
```

Comment:
Follow-up from above:

```suggestion

var numericTypeEffectiveInterfaceConformanceSet *InterfaceSet

func init() {
	numericTypeEffectiveInterfaceConformanceSet = NewInterfaceSet()
	numericTypeEffectiveInterfaceConformanceSet.Add(StructStringerType)
}

func (t *NumericType) EffectiveInterfaceConformanceSet() *InterfaceSet {
	return numericTypeEffectiveInterfaceConformanceSet
}
```